src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureNormalizers.java

```diff
@@ -0,0 +1,197 @@
+import com.o19s.es.ltr.feature.FeatureSet;
+import com.o19s.es.ltr.feature.NoOpFeatureNormalizerSet;
+import com.o19s.es.ltr.ranker.normalizer.NoOpNormalizer;
+import com.o19s.es.ltr.ranker.normalizer.Normalizer;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public class StoredFeatureNormalizers {
+
+
+    public enum Type implements Writeable {
+        STANDARD,
+        MIN_MAX;
+
+
+        public static Type readFromStream(StreamInput in) throws IOException {
+            int ord = in.readVInt();
+            for (Type type: Type.values()) {
+                if (type.ordinal() == ord) {
+                    return type;
+                }
+            }
+            throw new ElasticsearchException("unknown normalizer type during serialization [" + ord + "]");
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVInt(this.ordinal());
+        }
+    }
+
+    public static final ObjectParser.NamedObjectParser<FeatureNormDefinition, Void> PARSER;
+    private static final ParseField STANDARD = new ParseField("standard");
+    private static final ParseField MIN_MAX = new ParseField("min_max");
+
+    static {
+
+        PARSER = (XContentParser p, Void c, String featureName) -> {
+            // this seems really intended for switching on the key (here featureName) and making
+            // a decision, when in reality, we want to look a layer deeper and switch on that
+            ObjectParser<FeatureNormConsumer, Void> parser = new ObjectParser<>("feature_normalizers",
+                                                                                  FeatureNormConsumer::new);
+
+            BiConsumer<FeatureNormConsumer, StandardFeatureNormDefinition> setStd = (s, v) -> {
+                v.setFeatureName(featureName);
+                s.setFtrNormDefn(v);
+            };
+
+            BiConsumer<FeatureNormConsumer, MinMaxFeatureNormDefinition> setMinMax = (s, v) -> {
+                v.setFeatureName(featureName);
+                s.setFtrNormDefn(v);
+            };
+
+            parser.declareObject(setStd, StandardFeatureNormDefinition.PARSER, STANDARD);
+            parser.declareObject(setMinMax, MinMaxFeatureNormDefinition.PARSER, MIN_MAX);
+
+            FeatureNormConsumer parsedNorm  = parser.parse(p, c);
```

Comment:
Instead of Void for the context you could switch to String here and pass featureName this there's no need to set featureName after the fact and you don't need the two BiConsumer above as you may call FeatureNormConsumer::setFtrNormDefn directly I believe